src/api/procedures/rotatePrimaryKeyToSecondary.ts

```diff
@@ -0,0 +1,130 @@
+import { createAuthorizationResolver } from '~/api/procedures/utils';
+import { PolymeshError, Procedure } from '~/internal';
+import {
+  Authorization,
+  AuthorizationRequest,
+  AuthorizationType,
+  ErrorCode,
+  RotatePrimaryKeyToSecondaryParams,
+  TxTags,
+} from '~/types';
+import { ExtrinsicParams, TransactionSpec } from '~/types/internal';
+import {
+  authorizationToAuthorizationData,
+  expiryToMoment,
+  permissionsLikeToPermissions,
+  signerToSignatory,
+  signerToString,
+} from '~/utils/conversion';
+import { asAccount } from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export async function prepareRotatePrimaryKeyToSecondary(
+  this: Procedure<RotatePrimaryKeyToSecondaryParams, AuthorizationRequest>,
+  args: RotatePrimaryKeyToSecondaryParams
+): Promise<TransactionSpec<AuthorizationRequest, ExtrinsicParams<'identity', 'addAuthorization'>>> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: {
+          identity: { addAuthorization },
+        },
+      },
+    },
+    context,
+  } = this;
+  const { permissions, targetAccount, expiry } = args;
+
+  const issuerIdentity = await context.getSigningIdentity();
+
+  const target = asAccount(targetAccount, context);
+
+  const [authorizationRequests, targetIdentity, primaryAccount] = await Promise.all([
+    issuerIdentity.authorizations.getSent(),
+    target.getIdentity(),
+    issuerIdentity.getPrimaryAccount(),
+  ]);
+
+  const pendingAuthorization = authorizationRequests.data.some(authorizationRequest => {
+    const {
+      target: targetSigner,
+      data: { type },
+    } = authorizationRequest;
+    return (
+      signerToString(targetSigner) === targetAccount &&
+      !authorizationRequest.isExpired() &&
+      type === AuthorizationType.RotatePrimaryKeyToSecondary
+    );
+  });
+
+  if (pendingAuthorization) {
+    throw new PolymeshError({
+      code: ErrorCode.NoDataChange,
+      message:
+        'The target Account already has a pending invitation to become the primary key of the given Identity',
+      data: {
+        pendingAuthorization,
+      },
+    });
+  }
+
+  if (targetIdentity && issuerIdentity.isEqual(targetIdentity)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'The target Account is assigned to another Identity',
+      data: {
+        pendingAuthorization,
```

Comment:
`pendingAuthorization` is from `some` return and is always going to be false.
      
```suggestion
  targetDid: targetIdentity.did
```